mirrord/layer/src/exec_utils.rs

```diff
@@ -171,14 +171,40 @@ fn intercept_tmp_dir(argv_arr: &Nul<*const c_char>) -> Detour<Argv> {
     Success(c_string_vec)
 }
 
+/// Verifies that mirrord environment is passed to child process
+fn intercept_environment(envp_arr: &Nul<*const c_char>) -> Detour<Argv> {
+    let mut c_string_vec = Argv::default();
+
+    let mut found_dyld = false;
+    for arg in envp_arr.iter() {
+        let arg_str: &str = arg.checked_into()?;
```

Comment:
I think here instead of returning a `Detour::Bypass` if an env var is not valid UTF-8, we could just add it to `c_string_vec` and `continue` to the next env var, using `let ... else`. This will mean that we can still find the dyld var even if some other var is not a valid UTF-8 string.
The one limitation we would still have is that if for some reason the dyld var's value is not a valid UTF-8 string (not because of our lib, but because of something the user or their tools put in there), we would fail, but I guess that's ok for now?